### DISCUSSION TITLE
atom garbage collection - If all subscribed components is unmount, the atom will be garbage collected?

### DISCUSSION THREAD

**Discussion by euijinkk** [pmndrs/jotai#1587] [Q&A] [answered]
I know jotai uses WeakMap to define global state. If the atom is not referenced, it will be target of garbage collector by Javascript engine. Right?

But I wonder the relationship with react component lifecycle.
If all subscribed components are unmount, it will be target of garbage collector?

You know, a global state often needs to be kept although a component is unmount.
I wonder how jotai resolve it.

**Comment by dai-shi** [comment, marked as answer]
You are right. If atoms lost reference, their values will be garbage collected. If atoms are mounted, their references are kept in React, so not GC'd.

**Comment by euijinkk** [comment]
I test the code.
https://codesandbox.io/s/focused-matan-5s2yzc?file=/src/App.tsx

I think if all components that subscribe the atom are unmount, the atom will be GC'd.
In that code, I thought if Count component is unmount, countAtom value will be GC'd.
But that's not. 

Can you explain me something I don't understand?

**Reply by dai-shi** [reply]
```js
export const countAtom = atom(0);
```
That's still in memory.

You need to delete the reference too.
```js
export let countAtom = atom(0);

// then this will allow GC.
countAtom = null;
```

**Reply by euijinkk** [reply]
Aha,, 
Atom is reference value and it is key of WeakMap, so it should be change to be GC. 
I understand it.
Thank you dai-shi!!
